src/Data/Array/ST.purs

```diff
@@ -66,6 +68,11 @@ withArray f xs = do
 unsafeFreeze :: forall a r h. STArray h a -> Eff (st :: ST h | r) (Array a)
 unsafeFreeze = pure <<< (unsafeCoerce :: STArray h a -> Array a)
 
+-- | O(1) Convert an immutable array to a mutable array, without copying. The mutable
+-- | array must not be mutated.
```

Comment:
I think writing "the mutable array must not be mutated" seems to defeat the point of this function: if you have an `Array a` and you want to convert it into an `STArray h a`, that's probably because you want to mutate it, right? I think it would make more sense to say "the input array must not be used afterwards".

Reply:
Right. Will fix.